MailCore/Models/Signature.swift

```diff
@@ -60,9 +56,34 @@ public class Signature: Object, Codable, Identifiable {
         // Property 'hash' already exists
         case hashString = "hash"
     }
+
+    override public var hash: Int {
+        id
+    }
+
+    override public func isEqual(_ object: Any?) -> Bool {
+        guard let object = object as? Self else {
+            return false
+        }
+
+        return id == object.id
+    }
+}
+
+public extension Array where Element == Signature {
+    /// Find the default signature, if any, in  an `Array` of `Signature`
+    var defaultSignature: Signature? {
+        guard let defaultSignature = first(where: \.isDefault) else {
+            // We try to return at least a signature, so the backend is happy. Same on Android.
+            return first
+        }
+
+        // We matched one
+        return defaultSignature
+    }
 }
 
-public class ValidEmail: Object, Decodable {
+public final class ValidEmail: Object, Decodable {
```

Comment:
This object is not used can you remove it ? (I think it has not been used for quite some time now)